cmd/ls.go

```diff
@@ -43,28 +56,41 @@ func ls(cmd *cobra.Command, args []string) (err error) {
 	path := ""
 	if len(args) > 0 {
 		if path, err = validatePath(args[0]); err != nil {
-			return
+			return err
 		}
 	}
 
 	arg := files.NewListFolderArg(path)
 
 	res, err := dbx.ListFolder(arg)
+	var entries []*files.Metadata
 	if err != nil {
-		return
-	}
-
-	entries := res.Entries
-
-	for res.HasMore {
-		arg := files.NewListFolderContinueArg(res.Cursor)
+		// Don't treat a "not_folder" error as fatal; recover by sending a
+		// get_metadata request for the same path and using that response instead.
+		if strings.Contains(err.Error(), "path/not_folder/") {
```

Comment:
Overall LG. The SDK actually does have typed errors (e.g. `files.ListFolderError`) but I realize it is actually impossible to get at that from the returned error right now. Let me see if I can fix up the SDK. Otherwise we can go ahead with this as-is and I can fix it up later.